### PR TITLE
Add getValidAccessToken method to ConnectSdk

### PR DESCRIPTION
**Pull request by jorunfa**
* AccessTokenCallback added. This is callback that gives you an Access Token as a string or an error.
* There are now `ConnectTokens` and `ConnectTokensTO`. The latter is the transfer object. The first can now tell you if the Access Token is expired or not.

### CODE REVIEW

File: connect/src/com/telenor/connect/id/TokenStore.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public void update(ConnectTokens connectTokens) {
 
     public ConnectTokens get() {
         String connectTokensJson = context
-                .getSharedPreferences(PREFERENCES_FILE, Context.MODE_PRIVATE)
+                .getSharedPreferences(PREFERENCES_FILE,  Context.MODE_PRIVATE)
```
Comment (ulfcarlin) [inline connect/src/com/telenor/connect/id/TokenStore.java]: Double space => unnecessary change

File: connect/src/com/telenor/connect/ConnectRefreshTokenMissingException.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.telenor.connect;
+
+public class ConnectRefreshTokenMissingException extends IllegalStateException {
```
Comment (ulfcarlin) [inline connect/src/com/telenor/connect/ConnectRefreshTokenMissingException.java]: Are we not using checked exceptions? Is this an unrecoverable condition? Is this case really an illegal state?

File: connect/src/com/telenor/connect/id/ConnectIdService.java [hunk 3]
```diff
@@ -34,6 +34,19 @@ public ConnectIdService(
         this.tokenStore = tokenStore;
     }
 
+    public void getValidAccessToken(final AccessTokenCallback callback) {
+        if (retrieveTokens() == null) {
+            throw new ConnectRefreshTokenMissingException("Failed to call retrieveTokens().");
```
Comment (ulfcarlin) [inline connect/src/com/telenor/connect/id/ConnectIdService.java]: Is more than the refresh token missing? Could the message at least be improved?

File: connect/src/com/telenor/connect/ConnectRefreshTokenMissingException.java (hunk 2 above)
Comment (jorunfa) [inline connect/src/com/telenor/connect/ConnectRefreshTokenMissingException.java]: Hmm, bit unsure here. If the user is signed in but we don't have (refresh) tokens, then something is very wrong with the state, I'd say.

File: connect/src/com/telenor/connect/id/ConnectIdService.java (hunk 3 above)
Comment (jorunfa) [inline connect/src/com/telenor/connect/id/ConnectIdService.java]: Yes. Better now?

File: connect/src/com/telenor/connect/id/ConnectTokens.java [hunk 4]
```diff
@@ -1,46 +1,38 @@
 package com.telenor.connect.id;
 
-import com.google.gson.annotations.SerializedName;
+import com.telenor.connect.utils.Validator;
 
-public class ConnectTokens {
-    @SerializedName("access_token")
-    private final String accessToken;
+import java.util.Calendar;
+import java.util.Date;
```
Comment (ulfcarlin) [inline connect/src/com/telenor/connect/id/ConnectTokens.java]: Consider using java.time on Java 8, or Joda time otherwise?

File: connect/src/com/telenor/connect/ConnectRefreshTokenMissingException.java (hunk 2 above)
Comment (jorunfa) [inline connect/src/com/telenor/connect/ConnectRefreshTokenMissingException.java]: Extend simply from Exception instead?

File: connect/src/com/telenor/connect/id/IdToken.java [hunk 5]
```diff
@@ -60,6 +64,10 @@ public String getSubject() {
         return subject;
     }
 
+    public Date getExpirationDate() {
```
Comment (ulfcarlin) [inline connect/src/com/telenor/connect/id/IdToken.java]: Is this relevant to this commit? Was it mentioned in the commit message?

File: connect/src/com/telenor/connect/id/ConnectIdService.java [hunk 6]
```diff
@@ -36,7 +36,8 @@ public ConnectIdService(
 
     public void getValidAccessToken(final AccessTokenCallback callback) {
         if (retrieveTokens() == null) {
-            throw new ConnectRefreshTokenMissingException("Failed to call retrieveTokens().");
+            throw new ConnectRefreshTokenMissingException("retrieveTokens() returned null. Tokens" +
+                    "are missing.");
```
Comment (ulfcarlin) [inline connect/src/com/telenor/connect/id/ConnectIdService.java]: Missing space. Consider inserting the newline before the string literal instead, so that you do not need to split it up. I still think it is confusing to throw "ConnectRefreshTokenMissingException" when the client is trying to get an access token

File: connect/src/com/telenor/connect/ConnectSdk.java [hunk 7]
```diff
@@ -117,6 +118,20 @@ public static synchronized void authenticate(Activity activity,
         activity.startActivityForResult(intent, requestCode);
     }
 
+    /**
+     * Get a valid Access Token. If a non-expired one is available, that will be given to the
+     * callback {@code onSuccess(String accessToken)} method.
```
Comment (ulfcarlin) [inline connect/src/com/telenor/connect/ConnectSdk.java]: Consider mentioning that the token may expire or become revoked before being used. If so, the client should try to refresh the tokens explicitly.

File: connect/src/com/telenor/connect/id/IdToken.java (hunk 5 above)
Comment (jorunfa) [inline connect/src/com/telenor/connect/id/IdToken.java]: Maybe not. I can remove it from this commit.
